lib/threema/receive/delivery_receipt.rb

```diff
@@ -1,12 +1,46 @@
 # frozen_string_literal: true
 
+require 'threema/util'
+
 class Threema
   module Receive
     class DeliveryReceipt
-      attr_reader :content
+      STATUS_TYPE_BYTE = {
+        received: "\x01".b,
+        read: "\x02".b,
+        explicitly_acknowledged: "\x03".b,
+        explicity_declined: "\x04".b
+      }.freeze
+      UNHEXIFIED_MESSAGE_ID_LENGTH = 8
+
+      attr_reader :content, :status, :timestamp, :message_ids
 
       def initialize(content:, **)
         @content = content
+        return unless type_by(content).present?
+
+        @status = type_by(content)
+        @timestamp = Time.now.utc.to_i
+        @message_ids = extract_message_ids(content)
+      end
+
+      private
+
+      def type_by(content)
+        STATUS_TYPE_BYTE.key(content.slice(0))
+      end
+
+      def extract_message_ids(content)
```

Comment:
This could also be a lazy-loaded instance attribute, no? Called `message_ids`. Just an idea.

Reply:
Discussed this together and decided to process them on intialization.